src/components/ToDoList/ToDoList.tsx

```diff
@@ -0,0 +1,242 @@
+import React from 'react';
+import { useState } from 'react';
+import Button from '../Button/Button';
+import './ToDoList.css';
+import { ITodoList } from '../../App';
+import { MAX_ITEM_LENGTH } from '../constants';
+import Input from '../Button/Input/input';
+
+interface ITodos {
+  todos: ITodoList[];
+  setTodos: React.Dispatch<React.SetStateAction<ITodoList[]>>;
+}
+enum Filters {
+  All = 'all',
+  Active = 'active',
+  Completed = 'completed',
+}
+
+export default function ToDoList({ todos, setTodos}: ITodos) {
+  const [inputValue, setInputValue] = useState<string>('');
+  const [inputError, setInputError] = useState<boolean>(false);
+  const [showTextEror, setShowTextError] = useState<string>('');
+  const [disableStatusButton, setDisableStatusButton] = useState<boolean>(todos.length == 0 ? true : false);
+  const [filter, setFilter] = useState<Filters>(Filters.All);
+  const [editingItem, setEditingItem] = useState<number | null>(null);
+  const [inputValueEditing, setInputValueEditing] = useState<string>('');
+  const [inputEditError, setInputEditError] = useState<boolean>(false);
+
+  function addTask(): void {
+    if (inputValue.trim() !== '' && inputValue.length <= MAX_ITEM_LENGTH) {
+      setTodos([
+        ...todos,
+        { index: Math.random(), title: inputValue, isCompleted: false },
+      ]);
+      if (todos.length === 0) {
+        setDisableStatusButton(false);
+      }
+    } else {
+      setInputError(true);
+      if (inputValue.trim() === '') {
+        setShowTextError('*You must write something!');
+      } else {
+        setShowTextError('*Your ToDo is too long, please shorten it!');
+      }
```

Comment:
move to constants.ts 

TEXT_ERRORS = {
  blank: '*You must write something!',
  maxLength: '*Your ToDo is too long, please shorten it!',
}

```suggestion
        setShowTextError(TEXT_ERRORS.blank);
      } else {
        setShowTextError(TEXT_ERRORS.maxLength);
      }
```

Reply:
Done